2-Draft-Accepted/RFCNNNN-Support-Experimental-Features.md

```diff
@@ -89,6 +93,8 @@ PowerShell reads the experimental feature list from the configuration file only
   and read the list if not yet.
   This is to make sure experimental features can be enabled when an application is hosting PowerShell.
 
+> NOTE: Current command-line parsing may make some experimental features difficult to flag due to the amount of work that occurs before parsing is completed. This is seen on Linux with Logging overrides.  It may be necessary to move parsing earlier in the startup call-path.
```

Comment:
We should consider moving command-line parsing as early in startup as possible. For the case where the CommandLineParser is introducing an experimental feature, it could internally parse a second time if the experimental feature is enabled by the settings file.  This may require the CommandLine parser to populate the hashset or build the hash set and populate it at the end of parsing.  Finally, for the few cases where updates are made as a side-effect of parsing, such as with log settings, these updates should be performed after parsing completes.  In short, distinguishing between parse an apply.